File: CS/FilterDuplicateRecords_Unbound/Item.cs
using DevExpress.Mvvm;
using System;
using System.Collections.Generic;

namespace FilterDuplicateRecords_Unbound
{
    public class Item : BindableBase
    {
        public string Name { get; }

        public decimal Growth { get; set; }

        public Item(string name, decimal growth) {
            Name = name;
            Growth = growth;
        }

        public static IEnumerable<Item> GetData(int quantity) {
            var gen = new Random(42);

            for (int i = 0; i < quantity; i++) {
                var name = $"Name# {i}";
                var growth = 1 + 1m / gen.Next(1, 1000);

                yield return new Item(name, growth);
            }
        }
    }
}

File: VB/FilterDuplicateRecords_Unbound/Item.vb
Imports DevExpress.Mvvm
Imports System
Imports System.Collections.Generic

Namespace FilterDuplicateRecords_Unbound

    Public Class Item
        Inherits BindableBase

        Public ReadOnly Property Name As String

        Public Property Growth As Decimal

        Public Sub New(ByVal name As String, ByVal growth As Decimal)
            Me.Name = name
            Me.Growth = growth
        End Sub

        Public Shared Iterator Function GetData(ByVal quantity As Integer) As IEnumerable(Of Item)
            Dim gen = New Random(42)
            For i As Integer = 0 To quantity - 1
                Dim name = $"Name# {i}"
                Dim growth = 1 + 1D / gen.Next(1, 1000)
                Yield New Item(name, growth)
            Next
        End Function
    End Class
End Namespace
